api/app/weather_models/rdps_filename_marshaller.py

```diff
@@ -33,6 +35,25 @@ class SourcePrefix(enum.Enum):
 FORECAST_HOURS = [f"{hour:03d}" for hour in list(range(0, 84))]
 
 
+@dataclass(frozen=True)
+class WeatherModelKeyParams:
+    variable: str
+    level_type: str
+    level: str
+
+
+weather_key_parameters = {
+    "temp": WeatherModelKeyParams("TMP", "TGL", "2"),
+    "precip": WeatherModelKeyParams("APCP", "SFC", "0"),
+    "wind_speed": WeatherModelKeyParams("WIND", "TGL", "10"),
+    "rh": WeatherModelKeyParams("RH", "TGL", "2"),
+}
+
+
+def get_weather_key_params(parameter):
```

Comment:
No need to do this but just an aside: since this is a dict of `str -> WeatherModelKeyParams` the lookup will return `None` if the wrong string is passed. An alternative way to keep all the fields of `WeatherModelKeyParams` but enforce a type could be to subclass `Enum` with a dataclass mixin: https://docs.python.org/3/howto/enum.html#dataclass-support

Reply:
Thanks for the info!